src/camera.py

```diff
@@ -3,209 +3,182 @@
+        self._pan_servo = Servo.Servo(self.PAN_CHANNEL,
+                                      bus_number=bus_number,
                                       offset=self.OFFSET_TO_SERVO)
-        self._tilt_servo = Servo.Servo(self.TILT_CHANNEL, 
-                                       bus_number=bus_number, 
+        self._tilt_servo = Servo.Servo(self.TILT_CHANNEL,
+                                       bus_number=bus_number,
                                        offset=self.OFFSET_TO_SERVO)
 
         self._current_pan_angle = 0
         self._current_tilt_angle = 0
 
 
-    def set_pan(self, degree):
+    def pan(self, degree):
         """
         Pans the camera.
 
-        Controls the horizontal movement of the camera in [0, 180] degrees. 
-        0 refers to the vehicle's right while 180 refers to the vehicle's left. 
+        Controls the horizontal movement of the camera in [0, 180] degrees.
+        0 refers to the vehicle's right while 180 refers to the vehicle's left.
+        If given below 0, the camera will pan to 0. If given above 180, the
+        camera will pan to 180.
 
-        :param degree: The degree to set the pan servo to.
-        :type degree: float
+        :param degree: The number of degrees.
+        :type degree: int
         """
-        self._current_pan_angle = self.validate_degree(degree, 
-                                                       self.MIN_DEGREE,
-                                                       self.MAX_PAN_DEGREE)
-        self._pan_servo.write(self.validate_degree(self._current_pan_angle + 
-                                                   self._pan_offset,
-                                                   self.MIN_DEGREE,
-                                                   self.MAX_PAN_DEGREE))
+        self._current_pan_angle = numpy.clip(degree, self.MIN_DEGREE,
+                                                     self.MAX_PAN_DEGREE)
+        self._pan_servo.write(self._current_pan_angle + self._pan_offset)
 
 
     def calibrate_pan(self, degree):
         """
-        Calibrates camera pan.
+        Calibrates the camera's panning servo.
 
-        Controls offset in [-180, 180] degrees for pan. Writes new offset value to
-        'SunFounder_PiCar-V/remote_control/remote_control/driver/config' 
-        to 'pan_offset'. 
+        Writes new offset value for pan to the config file and changes current
+        pan offset to degree parameter.
 
-        :param degree: The amount of degrees to add or subtract for calibration.
-        :type degree: float
+        :param degree: The number of degrees.
+        :type degree: int
         """
-        self._pan_offset = self.validate_degree(degree, self.MIN_CALI_DEGREE,
-                                                        self.MAX_CALI_DEGREE)
-        self._DB.set('pan_offset', self._pan_offset)
+        self._pan_offset = int(numpy.clip(degree, self.MIN_CALI_DEGREE,
```

Comment:
If the user inputs a float value it will be saved, the system won't read it correctly when pulling the data. As long as the user obeys the param type there is no problem, I'm not sure what level of problem prevention we are running at.

Reply:
Sure